coreneuron/io/output_spikes.cpp

```diff
@@ -149,8 +149,58 @@ void sort_spikes(std::vector<double>& spikevec_time, std::vector<int>& spikevec_
     local_spikevec_sort(svt_buf, svg_buf, spikevec_time, spikevec_gid);
 }
 
-// Write generated spikes to out.dat using mpi parallel i/o.
-void output_spikes_parallel(const char* outpath, const std::string& population_name) {
+#ifdef ENABLE_SONATA_REPORTS
+/** Split spikevec_time and spikevec_gid by populations
+ *  Add spike data with population name and gid offset tolibsonatareport API
+ */
+void output_spike_populations(
+    const std::vector<std::pair<std::string, int>>& population_name_offset) {
+    // Write spikes with default population name and offset
+    if (population_name_offset.empty()) {
+        sonata_add_spikes_population("All",
+                                     0,
+                                     spikevec_time.data(),
+                                     spikevec_time.size(),
+                                     spikevec_gid.data(),
+                                     spikevec_gid.size());
+        return;
+    }
+    int n_populations = population_name_offset.size();
+    for (int idx = 0; idx < n_populations; idx++) {
+        auto cur_pop = population_name_offset[idx];
+        std::string population_name = cur_pop.first;
+        int population_offset = cur_pop.second;
+        int gid_lower = population_offset;
+        int gid_upper = -1;
+        if (idx != n_populations - 1) {
+            gid_upper = population_name_offset[idx + 1].second - 1;
+        }
+        std::vector<double> pop_spikevec_time;
+        std::vector<int> pop_spikevec_gid;
+        for (int j = 0; j < spikevec_gid.size(); j++) {
+            if (spikevec_gid[j] < gid_lower || spikevec_gid[j] > gid_upper && gid_upper >= 0) {
```

Comment:
I would inverse the if here, to insert only when the if condition is true:
if ( spikevec_gid[j] >= gid_lower && spikevec_gid[j] <= gid_upper ) {
    pop_spikevec_time.push_back(spikevec_time[j]);
    pop_spikevec_gid.push_back(spikevec_gid[j]);
}